packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -80,26 +85,43 @@ export default decorate([
           size: 'medium',
         })
 
-        markHubResourceAsInstalling(id)
-        try {
-          await Promise.all(
-            resourceParams.pools.map(pool =>
-              downloadAndInstallResource({
-                namespace,
-                id,
-                version,
-                sr: defined(
-                  resourceParams.mapPoolsSrs[pool.id],
-                  pool.default_SR
-                ),
-              })
+        const install = async () => {
+          markHubResourceAsInstalling(id)
+          try {
+            await Promise.all(
+              resourceParams.pools.map(pool =>
+                downloadAndInstallResource({
+                  namespace,
+                  id,
+                  version,
+                  sr: defined(
+                    resourceParams.mapPoolsSrs[pool.id],
+                    pool.default_SR
+                  ),
+                })
+              )
             )
-          )
-          success(_('hubImportNotificationTitle'), _('successfulInstall'))
-        } catch (_error) {
-          error(_('hubImportNotificationTitle'), _error.message)
+            success(_('hubImportNotificationTitle'), _('successfulInstall'))
+          } catch (_error) {
+            error(_('hubImportNotificationTitle'), _error.message)
+          }
+          markHubResourceAsInstalled(id)
+        }
+
+        const installedTemplates = filter(templates, [
+          'other.xo:resource:namespace',
+          namespace,
+        ])
+        const olderTemplates = filter(installedTemplates, template =>
```

Comment:
We can in the [installation process](https://github.com/vatesfr/xen-orchestra/pull/4613/files#diff-17032e40a62de5b80aa40071c01441faR93) delete the old templates when the installation succeed.